weave.lisp

```diff
@@ -197,71 +227,110 @@
-                  (setf (weaver-used-math weaver) t)
-                  (write-string "<div class=\"math-block\"><code>")
-                  (when (not (equal (second expr) "displaymath"))
-                      (format t "\\begin{~a}" (second expr)))
-                  (write-separated-list (third expr) #\newline *standard-output*)
-                  (when (not (equal (second expr) "displaymath"))
-                      (format t "\\end{~a}" (second expr)))
-                  (write-string "</code></div>"))
-                 (:MATH
-                  ; Use backticks to prevent markdown from formatting tex,
-                  ; for example treating _ as emphasis. 
-                  (setf (weaver-used-math weaver) t)
-                  (format t "<span class=\"math\">`~a`</span>"
-                          (second expr)))
-                 (:TOC (weave-toc
-                         (weaver-toc weaver)
-                         (textblockdef-file def)))
-                 (otherwise (error 'user-error
-                                   :format-control "unknown prose command ~S"
-                                   :format-arguments (first expr)))))
-              (t (error "unknown structure ~s" expr)))))
+    (cond ((stringp expr)
+           (if (ppcre:scan *anchor-pattern* expr)
+               (write-string
+                (replace-sections-with-id-anchors
+                 (weaver-section-map weaver)
+                 expr))
+               (write-string expr)))
+          ((commandp expr)
+           (case (first expr)
+             (:INCLUDE (weave-include
+                        (second expr)   ; title of block
+                        (textblockdef-file def)
+                        weaver
+                        nil))
+             (:TITLE
+              (setf (weaver-title weaver)
+                    (second expr)))
+             (:C
+                                        ; Also can act as a title
+              (when (null (weaver-title weaver))
+                (setf (weaver-title weaver)
+                      (second expr)))
+              (incf (weaver-chapter-counter weaver))
+              (setf (weaver-section-counter weaver) -1)
+              (format t "<h1>~a<a id=\"~a\"></a></h1>~%"
+                      (second expr)
+                      (chapter-id (weaver-chapter-counter weaver)))
+              (add-section weaver (list (second expr)) (chapter-id (weaver-chapter-counter weaver))))
+             (:S
+              (incf (weaver-section-counter weaver))
+              (format t "<h2>~a. ~a<a id=\"~a\"></a></h2>~%"
+                      (+ (weaver-section-counter weaver) 1)
+                      (second expr)
+                      (section-id
+                       (weaver-section-counter weaver)
+                       (weaver-chapter-counter weaver)))
+              (add-section weaver (list (second expr)) (section-id (weaver-section-counter weaver)
```

Comment:
Core piece #1 and #2 are here on line 289 and 298.

This seemed like a good place to hook in and save off the mapping of chapter/section titles to IDs.

Reply:
The table of contents `toc` structure on weave should have all the information you need. It it will specify what all the chapters and sections are and which files they are in. I would recommend making a pure function which takes the `toc` as input and produces whatever identifier trie you need.

Reply:
> The table of contents `toc` structure on weaver should have all the information you need.

Ah! That's obvious now.